### ISSUE TITLE
PHP Tests: Plugin integrations

### ISSUE THREAD

**Issue by swissspidy** [GoogleForCreators/web-stories-wp#5354]
## Task Description

Let's look into feasibility and usefulness of running the PHP test suite with other plugins active (Site Kit, AMP, etc.)

Worth testing that the AMP sanitization still works as expected when the AMP plugin is active 

Related: #5114, #5113

**Comment by spacedmonkey** [comment]
Doing some discovery on this ticket. I had the following thoughts. 

I think we should keep the currently unit test as is. We can add param like this. 
`test:php:amp-plugin` -> `vendor/bin/phpunit -c phpunit-amp-plugin.xml.dist` ( sudo code ).

 If the environment variable `AMP_PLUGIN_ENABLE` isset / true, then we can bootstrap in the amp like the like this. ( The amp plugin is already loaded in via composer ). 

tests/phpunit/bootstrap.php
```php
/**
 * Manually load the plugin being tested.
 */
function _manually_load_plugin() {
	require dirname( dirname( __DIR__ ) ) . '/web-stories.php';
        $_tests_amp = getenv( 'AMP_PLUGIN_ENABLE' );
        if ( $_tests_amp ) {
            if ( file_exists( WEBSTORIES_PLUGIN_DIR_PATH . '/includes/vendor/ampproject/amp-wp/amp.php' ) ) {
	        require WEBSTORIES_PLUGIN_DIR_PATH . '/includes/vendor/ampproject/amp-wp/amp.php';
            }
        }
}

tests_add_filter( 'muplugins_loaded', '_manually_load_plugin' );
```

We can then have test the just for when the amp plugin is enabled. It would work simple to multisite tests. Example test. 

```php
/**
 * @group amp-plugin-required
 */
public function test_init_sets_missing_options_amp() {
   ....
}
```

We could even have a group for skipped tests. 
```
@group amp-plugin-skipped
```

@swissspidy Do you have any ideas what what exactly you want to test. How is this effected by using the amp toolbox library?

**Comment by swissspidy** [comment]
I think this goes in the right direction 👍 

> Do you have any ideas what what exactly you want to test. How is this effected by using the amp toolbox library?

I was mostly interested in tests around the `Google\Web_Stories\Integrations\AMP` class, and making sure that the AMP plugin's logic is taking over from ours.

Ideally we build this in a way that we can easily add tests for other plugin integrations in the future if wanted

> `test:php:amp-plugin` -> `vendor/bin/phpunit -c phpunit-amp-plugin.xml.dist`

If we have an environment variable I don't think we need an extra config file.

> tests/phpunit/bootstrap.php

Note that `require WEBSTORIES_PLUGIN_DIR_PATH . '/includes/vendor/ampproject/amp-wp/amp.php';` will only work as long as we are including the amp-wp repo via Composer. This will change eventually (#4897 ).

> `@group amp-plugin-required`

I like it!

----

For reference, here is how the AMP plugin is doing it:

https://github.com/ampproject/amp-wp/blob/84a61f8165c021e364444983fb5caa431374f684/.github/workflows/build-test-measure.yml#L391-L392

https://github.com/ampproject/amp-wp/blob/84a61f8165c021e364444983fb5caa431374f684/.github/workflows/build-test-measure.yml#L391-L392

**Comment by spacedmonkey** [comment]
I think we would want it's own config file. So we have lines like this 

https://github.com/google/web-stories-wp/blob/ac28814481a9f65eb4c2cf58f36bb1c909c46afa/phpunit.xml.dist#L22-L26

**Comment by bmattb** [comment]
Needs more details from @swissspidy on what we agree on should be tested.

**Comment by swissspidy** [comment]
Related: #5473

**Comment by swissspidy** [comment]
I updated the ticket to reflect that this could potentially go beyond just the AMP plugin, adding integration tests for other plugins we have dedicated logic for.

For the AMP plugin, such tests could include verifying that our custom sanitizer is properly running, and the built-in sanitization is not kicking in, and that the various filters we add work as intended.

The above discovery nicely covers that, I think. Now the question is whether it's worth pursuing, or if the benefit (test coverage for a few integrations) does not outweigh the cost (increased maintenance, slower test runs)
